src/ess/reflectometry/tools.py

```diff
@@ -108,3 +112,160 @@ def linlogspace(
+                f(c),
+                left=np.nan,
+                right=np.nan,
+            )
+            for c in curves
+        ]
+    )
+
+
+def scale_reflectivity_curves_to_overlap(
+    curves: Sequence[sc.DataArray],
+    return_scaling_factors=False,
+) -> list[sc.DataArray] | list[sc.scalar]:
+    '''Make the curves overlap by scaling all except the first by a factor.
+    The scaling factors are determined by a maximum likelihood estimate
+    (assuming the errors are normal distributed).
+
+    All curves must be have the same unit for data and the Q-coordinate.
+
+    Parameters:
+        curves: the reflectivity curves that should be scaled together
+        return_scaling_factor:
+            If True the return value of the function
+            is a list of the scaling factors that should be applied.
+            If False (default) the function returns the scaled curves.
+
+    Returns:
+        A list of scaled reflectivity curves or a list of scaling factors.
+    '''
+    if len({c.data.unit for c in curves}) != 1:
+        raise ValueError('The reflectivity curves must have the same unit')
+    if len({c.coords['Q'].unit for c in curves}) != 1:
+        raise ValueError('The Q-coordinates must have the same unit for each curve')
+
+    qgrid = _create_qgrid_where_overlapping([c.coords['Q'] for c in curves])
+
+    r = _interpolate_on_qgrid(lambda c: c.data.values, curves, qgrid)
+    v = _interpolate_on_qgrid(lambda c: c.data.variances, curves, qgrid)
+
+    def cost(scaling_factors):
+        scaling_factors = np.concatenate([[1.0], scaling_factors])[:, None]
+        r_scaled = scaling_factors * r
+        v_scaled = scaling_factors**2 * v
+        v_scaled[v_scaled == 0] = np.nan
+        inv_v_scaled = 1 / v_scaled
+        r_avg = np.nansum(r_scaled * inv_v_scaled, axis=0) / np.nansum(
+            inv_v_scaled, axis=0
+        )
+        return np.nansum((r_scaled - r_avg) ** 2 * inv_v_scaled)
+
+    sol = opt.minimize(cost, [1.0] * (len(curves) - 1))
+    scaling_factors = (1.0, *sol.x)
+    if return_scaling_factors:
+        return [sc.scalar(x) for x in scaling_factors]
+    return [
+        scaling_factor * curve
+        for scaling_factor, curve in zip(scaling_factors, curves, strict=True)
+    ]
+
+
+def combine_curves(
+    curves: Sequence[sc.DataArray],
+    qgrid: sc.Variable | None = None,
+) -> sc.DataArray:
+    '''Combines the given curves by interpolating them
+    on a grid and merging them by the requested method.
+    The default method is a weighted mean where the weights
+    are proportional to the variances.
+
+    Unless the curves are already scaled correctly they might
+    need to be scaled using :func:`stitch_reflectivity_curves`.
+
+    All curves must be have the same unit for data and the Q-coordinate.
+
+    Parameters:
+        curves: the reflectivity curves that should be combined
+        qgrid: the Q-grid of the resulting combined reflectivity curve
```

Comment:
I think the syntax we have elsewhere is more like
```suggestion
    Parameters
    ----------
    curves:
        The reflectivity curves that should be combined
    qgrid:
        The Q-grid of the resulting combined reflectivity curve
```

Reply:
Yeah it's wrong, I'll fix that